WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -553,3 +565,62 @@ private extension MainTabBarController {
         viewModel.startObservingOrdersCount()
     }
 }
+
+// MARK: - Background Product Image Upload Status Updates
+
+private extension MainTabBarController {
+    func observeProductImageUploadStatusUpdates() {
+        guard featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) else {
+            return
+        }
+        productImageUploadStatusUpdatesSubscription = productImageUploader.statusUpdates.sink { [weak self] update in
+            self?.handleBackgroundImageUploadUpdate(update)
+        }
+    }
+
+    func handleBackgroundImageUploadUpdate(_ update: ProductImageUploadUpdate) {
+        if update.error != nil {
+            let notice = Notice(title: Localization.imageUploadFailureNoticeTitle,
+                                subtitle: nil,
+                                message: nil,
+                                feedbackType: .error,
+                                notificationInfo: nil,
+                                actionTitle: Localization.imageUploadFailureNoticeActionTitle,
+                                actionHandler: { [weak self] in
+                guard let self = self else { return }
+                Task { @MainActor in
+                    await self.showProductDetails(update: update)
+                }
+            })
+            noticePresenter.enqueue(notice: notice)
+        }
+    }
+
+    func showProductDetails(update: ProductImageUploadUpdate) async {
+        // Switches to the correct store first if needed.
+        let switchStoreUseCase = SwitchStoreUseCase(stores: stores)
+        let siteChanged = await switchStoreUseCase.switchStore(with: update.siteID)
+        if siteChanged {
+            let presenter = SwitchStoreNoticePresenter(siteID: update.siteID,
+                                                       noticePresenter: self.noticePresenter)
+            presenter.presentStoreSwitchedNoticeWhenSiteIsAvailable(configuration: .switchingStores)
+        }
+
+        let productViewController = ProductLoaderViewController(model: .product(productID: update.productID),
+                                                                siteID: update.siteID,
+                                                                forceReadOnly: false)
+        let productNavController = WooNavigationController(rootViewController: productViewController)
+        productsNavigationController.present(productNavController, animated: true)
+    }
+}
+
+private extension MainTabBarController {
+    enum Localization {
+        static let imageUploadFailureNoticeTitle =
+        NSLocalizedString("An image failed to upload",
```

Comment:
Just thinking out loud here,
If more than one image failed to upload within a second, we will still show a single notice with the text `"An image failed to upload"`([as Notice presenter avoids showing duplicate notices](https://github.com/woocommerce/woocommerce-ios/blob/fc47c8e0cd35dbb8bc06b7b44907c5d2915e2fdf/WooCommerce/Classes/Tools/Notices/DefaultNoticePresenter.swift#L35)) which conveys that only one image upload failed. 🤔 

But, considering that we don't have a way to display the failed image uploads, this is out of scope. 😀